### PR TITLE
refactor: Update snapshotting logic

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
This approach tries to align the way we take snapshots with custom attributes to what XCTest does. Also, if a snapshot with custom attributes fails then it is going to fallback to the default one after `customSnapshotTimeout` expires. This ensures we can always retrieve the same tree that XCTest, but does not ensure same values for custom attributes (like the `visible` one). It is also possible to completely disable custom attributes by setting `customSnapshotTimeout` to zero in order to avoid any potentially inconsistent results

### CODE REVIEW

**Review by Dan-Maor — APPROVED** [review]

File: WebDriverAgentLib/Utilities/FBSettings.m [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/**
```
Comment (KazuCocoa) [inline WebDriverAgentLib/Utilities/FBSettings.m:1]: nice 👍

**Review by KazuCocoa — COMMENTED** [review]
🆒 
Will catch up with recent refactors this weekend

**Comment by KazuCocoa** [comment]
Not sure by this or https://github.com/appium/WebDriverAgent/pull/404 , but I found a scroll scenario failed in https://github.com/appium/ruby_lib_core/blob/7801efa72b10ff3c601374af3e0e7dc68e3c8733/test/functional/ios/webdriver/w3c_actions_test.rb#L44 with the latest beta on CI. (I retried it a couple of times)
Will take a look at the cause probably this weekend.

**Comment by mykola-mokhnach** [comment]
Thanks for checking it @KazuCocoa 
The local integration tests all pass, so it would be interesting to know what went wrong there.

**Pull request merged**